src/net_processing.cpp

```diff
@@ -2018,6 +2046,21 @@ std::set<const CBlockIndex*>& PeerManagerImpl::GetChainTipSetForPeerNetwork(cons
     return m_chain_tip_sets.emplace(GetUniqueNetworkID(m_connman, node), std::set<const CBlockIndex*>()).first->second;
 }
 
+const CBlockIndex* PeerManagerImpl::LookupBlockIndexForPeer(const CNode& node, const uint256& hash)
+{
+    AssertLockHeld(cs_main);
+
+    const CBlockIndex* index{m_chainman.m_blockman.LookupBlockIndex(hash)};
+    if (!index) return nullptr;
+
+    if (IsAncestorOfBestHeaderOrTip(index) ||
```

Comment:
`IsAncestorOfBestHeaderOrTip` would already by contained in `IsBlockInChainTipSet` right? do we include `IsAncestorOfBestHeaderOrTip` check for performance gain instead of iterating over all the chain tips? or is there some other reason?